### PR TITLE
Structure Path Following section

### PR DESCRIPTION
**Pull request by Daltz333**
Separates the distinction between PathFinder and the newer path following classes.

### CODE REVIEW

File: source/docs/software/wpilib-tools/path-planning/index.rst [hunk 1]
```diff
@@ -1,13 +1,9 @@
-Autonomous Path Planning
-========================
+Path Following
+==============
+
+There are two main implementions of path-followering/trajectory generation. Pathfinder and the new WPILib trajectory classes.
```
Comment (AustinShalit) [inline source/docs/software/wpilib-tools/path-planning/index.rst]: We should provide some guidance here about what to use.

**Comment by Daltz333** [comment]
Thats a controversial opinion right now :sweat_smile:

**Comment by calcmogul** [comment]
WPILib traj gen is technologically better than PFv1 in every way. It's easier to use, supports constraints, and doesn't blow up on waypoints 90 degrees to each other. Idk what's so controversial.

File: source/docs/software/wpilib-tools/path-planning/pathfinder/index.rst [hunk 2]
```diff
@@ -0,0 +1,13 @@
+Autonomous Path Planning
```
Comment (AustinShalit) [inline source/docs/software/wpilib-tools/path-planning/pathfinder/index.rst]: Maybe Pathfinder should be in the title?

**Pull request merged**